### ISSUE TITLE
redirect to external site in root.tsx loader doesn't work, still calls App

### ISSUE THREAD

**Issue by cjkooij** [epicweb-dev/epic-stack#256]
Trying to return a redirect from the root.tsx loader doesn't redirect and tries to render the App.

See screenshots below, redirect is the only line added on a freshly installed epic-stack.

<img width="859" alt="image" src="https://github.com/epicweb-dev/epic-stack/assets/72307251/b345e231-39a3-4658-8303-036f809a1958">

<img width="1369" alt="image" src="https://github.com/epicweb-dev/epic-stack/assets/72307251/b56ff687-16f2-4f65-93b8-ac11dce20692">

**Comment by cjkooij** [comment]
throwing redirect does work, credit to @jacobparis

**Comment by kentcdodds** [comment]
I'm curious why returning a redirect doesn't work though. Perhaps this is a bug for Remix?

**Comment by cjkooij** [comment]
Weirdly enough if I do this in a plain stack it seems to work.

**Comment by kentcdodds** [comment]
Thanks for checking on that! We'll definitely want to figure out what it is then.

**Comment by onemen** [comment]
maybe the bug is in `server/index.js`

when i replace `server/index.js` file with the content of  [remix express/server](https://github.com/remix-run/remix/blob/main/templates/express/server.js)

with just one change
```diff
-  const BUILD_PATH = './build/index.js'
+  const BUILD_PATH = '../build/index.js'
```

the redirect work as expected

**Comment by onemen** [comment]
something wrong with sentry `wrapExpressCreateRequestHandler`, when removed from `server/index.js` redirect work as expected

```diff
import path from 'path'
import { fileURLToPath } from 'url'
import express from 'express'
import chokidar from 'chokidar'
import compression from 'compression'
import morgan from 'morgan'
import address from 'address'
import closeWithGrace from 'close-with-grace'
import helmet from 'helmet'
import crypto from 'crypto'
-import {
-  type RequestHandler,
-  createRequestHandler as _createRequestHandler,
-} from '@remix-run/express'
+import { type RequestHandler, createRequestHandler } from '@remix-run/express'
import { wrapExpressCreateRequestHandler } from '@sentry/remix'
import { type ServerBuild, broadcastDevReady } from '@remix-run/node'
import getPort, { portNumbers } from 'get-port'
import chalk from 'chalk'

// @ts-ignore - this file may not exist if you haven't built yet, but it will
// definitely exist by the time the dev or prod server actually runs.
import * as remixBuild from '../build/index.js'
const MODE = process.env.NODE_ENV

-const createRequestHandler = wrapExpressCreateRequestHandler(
-  _createRequestHandler,
-)

const BUILD_PATH = '../build/index.js'

**Comment by kentcdodds** [comment]
Hey @scefali, do you have any idea what may be causing this issue? Looks like it's only present when Sentry's wrapper is involved.

**Comment by scefali** [comment]
@kentcdodds I do not but I'm having someone take a look

**Comment by AbhiPrasad** [comment]
@onurtemizkan something lets take a look at?

This might be fixed by us merging and releasing https://github.com/getsentry/sentry-javascript/pull/8415

Sorry for the trouble folks!

**Comment by AbhiPrasad** [comment]
Also if you could run `npm why @sentry/remix` and report the version would help us debug a lot!

**Comment by AbhiPrasad** [comment]
We'll try and get a fix out friday or start of next week, thanks for your patience everyone.

**Comment by AbhiPrasad** [comment]
We've released a new version of `@sentry/remix` with the fix in [`7.58.0`](https://github.com/getsentry/sentry-javascript/releases/tag/7.58.0). Can you try updating your SDK version and seeing if the error persists? Thanks!

**Comment by kentcdodds** [comment]
Thanks a lot!
